### PR TITLE
fix: Fix runnable examples

### PR DESCRIPTION
**Pull request by lizMSFT**
Fix runnable examples by moving the const/variables into another file.

Reference: https://pkg.go.dev/go/doc#Example
> Playable Examples must be in a package whose name ends in "_test". An Example is "playable" (the Play field is non-nil) in either of these circumstances:
> - The example function is self-contained: the function references only identifiers from other packages (or predeclared identifiers, such as "int") and the test file does not include a dot import.
> - The entire test file is the example: the file contains exactly one example function, zero test, fuzz test, or benchmark function, and at least one top-level function, type, variable, or constant declaration other than the example function.

Using godoc to open the docs for local packages:
`go install -v golang.org/x/tools/cmd/godoc@latest`
`godoc -goroot . -play`

Resolves: #218 

Signed-off-by: Zoey Li <zoeyli@microsoft.com>

### CODE REVIEW

File: registry/example_const_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package registry_test gives examples code of functions in the registry package.
+package registry_test
```
Comment (wangxiaoxuan273) [inline registry/example_const_test.go:17]: Maybe` example_test_const.go` is a better filename? `example_const_test.go` is misleading.

File: registry/remote/auth/example_const_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const (
+	username     = "test_user"
+	password     = "test_password"
+	accessToken  = "test/access/token"
+	refreshToken = "test/refresh/token"
+)
+
+var (
+	host                  string
```
Comment (wangxiaoxuan273) [inline registry/remote/auth/example_const_test.go:25]: Maybe it's better for variables to remain in the test file, while putting the consts in the const file?

Comment (lizMSFT) [reply registry/remote/auth/example_const_test.go:25]: Some test cases did not use the constants we moved to the const file, they only use those variables. So if we put these variables back in the test file, the example will be playable then. Or do you have any suggestion?

File: registry/example_const_test.go (hunk 1 above)
Comment (lizMSFT) [reply registry/example_const_test.go:17]: Both files need to be end with `_test.go` in order to be successfully compiled.

**Comment by lizMSFT** [comment]
Hi @m5i-work , that's a brilliant idea! Thank you for your suggestion.

**Review by qweeah — APPROVED** [review]
LGTM

File: internal/playable/playable.go [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package playable
+
+const ExampleUnplayable = true
```
Comment (shizhMSFT) [inline internal/playable/playable.go:3]: Please document this.

File: internal/playable/playable.go [hunk 4]
```diff
@@ -0,0 +1,3 @@
+package playable
```
Comment (shizhMSFT) [inline internal/playable/playable.go:1]: Is `doc` a better package name?

Comment (shizhMSFT) [inline internal/playable/playable.go:1]: Missing license header.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: internal/playable/playable.go (hunk 3 above)
Comment (lizMSFT) [reply internal/playable/playable.go]: Added. Thanks for pointing this out.

File: internal/playable/playable.go (hunk 4 above)
Comment (lizMSFT) [reply internal/playable/playable.go]: Agree. Which one do you think is better, `doc` or `godoc`?

**Review by shizhMSFT — APPROVED** [review]
LGTM with suggestions

File: internal/playable/playable.go (hunk 4 above)
Comment (shizhMSFT) [reply internal/playable/playable.go]: It's a valid concern.

Comment (shizhMSFT) [reply internal/playable/playable.go]: How about moving the package to `registry/remote/internal/doc` so that the `doc` package is specific.

Comment (lizMSFT) [reply internal/playable/playable.go]: `registry/example_test.go` is also using this `ExampleUnplayable` value. If we move the package to `registry/remote/internal/doc`, will raise `invalid use of internal package` error in `registry/example_test.go`. How about moving to `registry/internal/doc`?

Comment (shizhMSFT) [reply internal/playable/playable.go]: > How about moving to `registry/internal/doc`?

That makes sense.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: internal/playable/playable.go (hunk 4 above)
Comment (lizMSFT) [reply internal/playable/playable.go]: Rebased and made requested change.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**